### ISSUE TITLE
[Q/A] "used-style" is simple ?

### ISSUE THREAD

**Issue by c0ncentus** [theKashey/used-styles#45]
I read that you scan html and css but maybe isnt so easy ...
and also put 'hybrid', server side rendering and whatever ... 

Can you tell me if you can expose a "main" function with html string (or whatever), css string (or whatever) as input 
and then have as output, a css used  ?

**Comment by theKashey** [comment]
>  a "main" function with html string (or whatever), css string (or whatever) as input
and then have as output, a css used ?

That is `getCriticalRules`, see https://github.com/theKashey/used-styles/blob/master/__tests__/extraction.spec.tsx#LL27C18-L27C18
Examples uses `loadStyleDefinitions` which is a little _in between_ of fully automated style discoverability and "just styles", you better use `parseProjectStyles`  see https://github.com/theKashey/used-styles/blob/632b8d2b0547e3a084f05b639a870ca1e3926cc5/__tests__/react.integration.spec.tsx#L134

That would be something like
```tsx
const styles = parseProjectStyles({
 style: `body { padding:0 }`
});

const critical = getCriticalRules('<HTML><body>hello</body></HTML>', styles);
```

This code works, but TypeScript will complain
- check this example - https://github.com/theKashey/used-styles/blob/master/__tests__/simple-example.spec.tsx#L3
- just released version 2.43 contains a fix for types

**Comment by c0ncentus** [comment]
ok thanks :)
